DataVirtualHardDisk: support source_media_link and __init__ args

melor opened the pull request:
Hi, I needed DataVirtualHardDisk source_media_link for testing and this seemed to work for my purposes. Did NOT run tests...

huguesv commented on azure/servicemanagement/__init__.py:
```diff
@@ -1353,13 +1353,16 @@ def __getitem__(self, index):
+        self.host_caching = host_caching
+        self.disk_label = disk_label
+        self.disk_name = disk_name
+        self.lun = lun
```
changing the default value of `lun` and `logical_disk_size_in_gb` from 0 to None is a breaking change.  the automatic xml deserialization knows to parse the xml as an integer if the default value is an integer, otherwise it will keep it as a string.  For example, when you query a virtual hard disk, you'll get a lun '1' instead of 1.  I know, it's a bad situation, and we are moving away from that kind of automatic xml deserialization for the next gen library.

melor commented:
Rebased and fixed the backward-compatibility breaking change (lun/logical_disk_size_in_gb now default to 0). Is None ok as a default for the other values, or do they need to be explicitly u''?

melor commented:
Oops, could have used immutable 0 directly as the default arg values. Can change that still, let me know if you want it that way. (Too paranoid about default values other than None after screwing it up with mutable [] and {})

huguesv commented:
Yeah, I've been lucky enough to be warned early on to never use [] and {} as default parameter values, so I never experienced that pain :)

There's a difference between u'' and None when it comes to *serializing* to xml.  None means the xml element won't appear at all, u'' means it will appear and be empty.  As far as *deserializing*, there's no difference.

It may be that None is more convenient for the user (opt in), otherwise you have to explicitly set the values you don't want to None, which isn't as obvious.

Changing that default from u'' to None has potential to break existing code though.  I would prefer you don't change any of them from what they used to be.  For the new one you are introducing, you can make that one None, in fact I think you have to, to avoid breaking existing code that doesn't know about that new field.

melor commented:
Default values now exactly same as before.
